src/sentry/replays/consumers/recording/process_recording.py

```diff
@@ -253,12 +61,43 @@ def __throttled_commit(self, force: bool = False) -> None:
                 self.__last_committed = now
                 self.__commit_data = {}
 
-    def _configure_sentry_scope(self, message_dict: dict[str, Any]) -> None:
-        with configure_scope() as scope:
-            scope.set_tag("replay_id", message_dict["replay_id"])
-            scope.set_tag("project_id", message_dict["project_id"])
-            # TODO: add replay sdk version once added
+
+def process_message(message: Message[KafkaPayload]) -> None:
+    try:
+        with sentry_sdk.start_transaction(
+            name="replays.consumer.process_recording",
+            op="replays.consumer",
+            sampled=random.random()
+            < getattr(settings, "SENTRY_REPLAY_RECORDINGS_CONSUMER_APM_SAMPLING", 0),
+        ):
+            message_dict = msgpack.unpackb(message.payload.value)
+
+            with configure_scope() as scope:
+                # TODO: add replay sdk version once added
+                scope.set_tag("replay_id", message_dict["replay_id"])
+                scope.set_tag("project_id", message_dict["project_id"])
+
+            if message_dict["type"] == "replay_recording_chunk":
+                process_chunk(cast(RecordingSegmentChunkMessage, message_dict))
+            elif message_dict["type"] == "replay_recording":
+                process_recording(cast(RecordingSegmentMessage, message_dict))
+    except Exception:
+        # avoid crash looping on bad messsages for now
+        logger = logging.getLogger("sentry.replays")
```

Comment:
i believe you can move the logger instantiation to the top of the file outside the function